lms/djangoapps/discussion_api/api.py

```diff
@@ -274,6 +277,7 @@ def get_thread_list(
         "last_activity_at".
     order_direction: The direction in which to sort the threads by. The only
         values are "asc" or "desc". The default is "desc".
+    discussion_id: Content-specific discussion topic ID
```

Comment:
Thank you @jimabramson and @robrap.  This is very helpful.
The question boiled down to what nomenclature to expose via the discussion API - whether to keep (inline) `discussion_id` as a separate concept from `topic_id`.  Given what you said and the given code, it looks like discussion_ids are _equivalent_ to topic_ids.  The distinction, however, is between a topic being "top-level" or "inline".
Is that correct?

If so, @wajeeha-khalid I agree with you.  We can just use the existing `topic_id_list` as the field for specifying the inline `discussion_id`.  And in so doing, we should change the field name returned by the discussion module's `student_view_data` method to be `topic_id`, so it's clear to clients that they are one and the same.